I113 generate csl citation

ShanaLMoore opened the pull request:
# Summary

When a user visits a record's show page, a CSL citation will be displayed in the sidebar. 

Issue: 
- https://github.com/scientist-softserv/west-virginia-university/issues/113
- https://github.com/scientist-softserv/west-virginia-university/issues/114

# Screenshots / Video

![Screenshot 2023-10-23 at 15-31-47 Congressman Dole's Introduction Of Senator Barry Goldwater - American Congress Digital Archives Portal](https://github.com/wvulibraries/hydra_acda_portal_public/assets/10081604/fc62a657-fd73-4cae-84b2-90f518f50901)

Commit:
✅ Set up rspec-rails

Create setup files for rspec by running `rails g rspec:install`.

Commit:
🎁 Implement Chicago Style Citation Service

In this commit a CSL citation service is implemented along with its base specs. c/o @kirkkwang

Issue:
- https://github.com/scientist-softserv/west-virginia-university/issues/113

Co-Authored-By: Kirk Wang <k3wang@gmail.com>

Commit:
🎁 Expose CSL citation on the frontend

In this commit we connect the frontend with the citation service so that the citation is rendered on the page when visiting a record.

Co-Authored-By: Kirk Wang <k3wang@gmail.com>

jeremyf commented on hydra/app/views/catalog/show.html.erb:
```diff
@@ -30,4 +30,16 @@
\ No newline at end of file
+  <div id="csl-citation">
+      <h4>Chicago citation style</h4>
+    <span class="chicago-citation"><%= ChicagoCitationService.new.format(@document, request.original_url) %></span>
```
Minor refinement; consider one of two options:

1. `ChicagoCitationService.format(document: @document, original_url: request.original_url)`
2. `ChicagoCitationService.new(document: @document, original_url: request.original_url).format`

I would expect to pass the "initial" values to either initialize or a single class method.  This narrows the interface of the class; and in option 1 could move to a module which has nominal memory improvements.  (Instantiating a class comes with a minor memory hit and garbage collection hit).

Otherwise, this is a fantastic feature!

kirkkwang replied:
Thanks for that, opted for option 1

laritakr commented on hydra/.rspec:
```diff
@@ -0,0 +1 @@
+--require spec_helper
```
Since specs can now be run, it would be helpful to have something about running specs added into the development.md documentation.

kirkkwang replied:
Cool, I added some basic documentation on running specs

jeremyf commented on hydra/app/views/catalog/show.html.erb:
```diff
@@ -30,4 +30,16 @@
\ No newline at end of file
+  <div id="csl-citation">
+      <h4>Chicago citation style</h4>
+    <span class="chicago-citation"><%= ChicagoCitationService.new.format(@document, request.original_url) %></span>
```
I'm not seeing a CSS selector for `.chicago-citation`; is this necessary?  Could this be a `p`-tag?

kirkkwang replied:
Good catch on that, using a `<p>` instead

jeremyf reviewed (approved):
My comments are non-blocking, just nudges.

Commit:
🧹 Refactor and update docs

This commit addresses the comments in the review.  Documentation on
running specs were added.  `ChicagoCitationService` was refactor to be
a module instead of a class and the specs and view has been updated to
to reflect the changes.
